test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -1810,3 +1811,27 @@ fn test_vector_with_capacity_u64() {
         Option::Some(val) => revert(0), Option::None => (), 
     }
 }
+
+fn test_vector_swap() {
+    let mut vector = ~Vec::new();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+    let number2 = 2u8;
+    let number3 = 3u8;
+
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+    vector.push(number3);
+
```

Comment:
add an assertion to prove that your vec is the size that you expect it to be and also capacity. after that you must check that your element 1 and element 3 are what you expect them to be. this is a "before condition" so before you use your method you must assert that your initial state is what you expect it to be otherwise your test might pass falsely because the implementation could be incorrect or there might be an underlying bug

Reply:
> add an assertion to prove that your vec is the size that you expect it to be and also capacity. after that you must check that your element 1 and element 3 are what you expect them to be. this is a "before condition" so before you use your method you must assert that your initial state is what you expect it to be otherwise your test might pass falsely because the implementation could be incorrect or there might be an underlying bug

Do we really need to assert vec size, capacity, element 1 and element 3 in "before condition" that they have proper initial values? I think it would be unnecessary as other tests in vec test suite should catch implementation errors of `Vec` initialization and push. What do you think? Wouldn't it be double checking same functionality?

Reply:
Yes you do. You must assert your initial condition. It doesn't matter how you got there but it matters that you can prove your initial condition. You cannot rely on other tests to help test your current test. That's coupling code.